service/eventer/github/client.go

```diff
@@ -44,35 +44,27 @@ func (e *Eventer) request(req *http.Request) (*http.Response, error) {
 	return resp, err
 }
 
-// filterDeploymentsByStatus filters out deployments that are finished -
+// filterDeploymentsWithoutStatuses filters out deployments that are finished -
 // that is, there exists at least one status that is not pending.
-func (e *Eventer) filterDeploymentsByStatus(deployments []deployment) []deployment {
+func (e *Eventer) filterDeploymentsWithoutStatuses(deployments []deployment) []deployment {
 	matches := []deployment{}
 
 	for _, deployment := range deployments {
 		// If there are any statuses apart from pending, we consider the
 		// deployment finished, and do not act on it.
-		isPending := true
-		for _, status := range deployment.Statuses {
-			if status.State != pendingState {
-				isPending = false
-				break
-			}
+		if len(deployment.Statuses) != 0 {
+			continue
```

Comment:
Now we filter everything that does not have any status. In the next steps the eventer will set a pending state to the deployment event and the operator will set a success or failure state.